test/libsolidity/semanticTests/inlineAssembly/optimize_memory_store_multi_block.sol

```diff
@@ -0,0 +1,23 @@
+contract Test {
+	uint256 x;
+
+	function test() public returns (uint256) {
+		uint256 a = myGetX();
+		x = 5;
+		uint256 b = myGetX();
+		assembly {
+			log0(0, 64)
+		}
+		return a + b + myGetX();
+	}
+
+	function myGetX() internal view returns (uint256) {
+		assembly {
+			mstore(1, 0x123456789abcdef)
+		}
+		return x;
+	}
```

Comment:
I think we had more affected cases on the channel. Maybe it would be a good idea to add them all?

Reply:
Added another test file with simpler cases. I don't think we need more than that, but if you have some in mind, I'll happily add.